### PR TITLE
Add basic food system with apple visual design

### PR DESCRIPTION
**Pull request by N3SSQwiK**
## Summary

Implements the complete food system per openspec proposal `add-basic-food`, completing the **minimal playable game milestone**. Players can now collect food to grow the snake and accumulate score.

### Features Implemented

| Feature | Description |
|---------|-------------|
| **Food Class** | Standalone class managing position, points, spawn logic, decay timing |
| **Random Spawning** | 100 random attempts, fallback to exhaustive valid cell search |
| **Decay System** | 10-second timer (100 ticks at 10Hz), respawns when expired |
| **Decay Warning** | Visual blink at <25% time remaining (every 0.5s) |
| **Collision Detection** | Triggers snake growth (+1 segment) and score increment (+10 pts) |
| **Apple Rendering** | Bezier curve apple shape with brown stem and green leaf |
| **Score Display** | Top-left corner, monospace font with shadow for readability |

### Food Class (`game.js:147-221`)

```javascript
class Food {
    constructor(gridWidth, gridHeight, decayTicks = FOOD_DECAY_TICKS)
    spawn(excludePositions, currentTick)  // Random with fallback
    checkCollision(headPosition)           // Boolean
    isExpired(currentTick)                 // Boolean
    isDecayWarning(currentTick)            // Boolean (< 25% remaining)
    reset()                                // Clear position and timer
}
```

**Spawn Algorithm:**
1. Try 100 random positions, checking against snake body
2. If all fail, compute all valid cells and pick randomly
3. Returns `false` if grid is completely full (edge case)

### Renderer Methods (`game.js:412-453`)

- `drawFood(food, isDecayWarning, currentTick)` - Apple shape using bezier curves
- `drawScore(score, length)` - Text with drop shadow

### Game Integration (`game.js:625-769`)

- **Constructor**: Initializes Food instance, score=0, tickCount=0
- **tick()**: Spawns food, checks collision (grow + score + respawn), checks decay
- **render()**: Draws food (with blink state) and score during PLAYING/PAUSED/GAMEOVER
- **reset()**: Clears score, tickCount, and food position

## Design Decisions

Reviewed with `/game-ui-design` skill per CLAUDE.md requirements:

| Element | Choice | Rationale |
|---------|--------|-----------|
| Shape | Apple (bezier curves) | Recognizable food iconography, contrasts with angular snake |
| Color | `theme.colors.food` | Theme-aware, not hardcoded |
| Glow | 8px blur, food color | Draws attention without neon overload |
| Stem | Brown (#5d4037) | Natural apple appearance |
| Leaf | Green (#4caf50) | Adds recognizability |
| Decay Warning | Blink toggle every 5 ticks | Clear urgency signal |

See `openspec/changes/archive/2026-01-28-add-basic-food/design-decisions.md` for full documentation.

## Test Coverage

**111 tests total** (39 new for food system):

### Food Class Tests
- Constructor initialization and custom decay ticks
- Spawn in valid position, avoiding excluded positions
- Spawn fallback when grid is nearly full
- Spawn returns false when grid is completely full

### Collision Tests
- Match when positions equal
- No match when positions differ
- Safe handling of null position

### Decay Tests
- `isExpired` timing (before, at, after timeout)
- `isDecayWarning` threshold (>25%, =25%, <25%)
- Reset clears position and spawnTick

### Game Integration Tests
- Game has food instance, score=0, tickCount=0
- Tick spawns food if none exists
- Eating increments score, grows snake, respawns food
- Reset clears score, tickCount, and food
- Food respawns when expired
- Score visible during GAMEOVER state
- Food NOT drawn during GAMEOVER state

### Renderer Tests
- `drawFood` draws apple shape (bezier curves + ellipse)
- Skips drawing when position is null
- Blink effect toggles visibility during decay warning
- `drawScore` sets correct font properties

## Edge Cases Handled

| Edge Case | Solution |
|-----------|----------|
| Grid nearly full | Fallback to exhaustive valid cell search |
| Grid completely full | `spawn()` returns false, no crash |
| Rapid food collection | Score increments correctly, growth queued |
| Decay during pause | Timer paused (uses tick count, not wall time) |
| Game over | Score remains visible, food hidden |

## Out of Scope (Separate Proposals)

- **Wall collision** → `openspec/changes/add-wall-collision/`
- **Bonus/poison food** → Future proposal (theme already defines colors)
- **Game over screen** → `openspec/changes/add-ui-screens/`

## How to Test

1. Run unit tests: `node --test game.test.js`
2. Browser test:
   - `npx serve .` and open http://localhost:3000
   - Food appears as red apple (not on snake)
   - Eating: score +10, snake grows, food respawns elsewhere
   - Wait 10s: food blinks at ~7.5s, respawns at 10s
   - Collide with self: game stops, score visible

## Checklist

- [x] All 40 tasks from proposal completed
- [x] 111 unit tests pass
- [x] Design reviewed with `/game-ui-design` skill
- [x] Proposal archived to `openspec/changes/archive/2026-01-28-add-basic-food/`
- [x] Spec created at `openspec/specs/food-system/spec.md`
- [x] Manual browser testing completed

---

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: game.js [hunk 1]
```diff
@@ -134,6 +140,86 @@ class Snake {
...
+
+        if (validCells.length > 0) {
+            const randomIndex = Math.floor(Math.random() * validCells.length);
+            this.position = validCells[randomIndex];
+            this.spawnTick = currentTick;
+            return true;
+        }
+
+        // No valid position available (grid is full)
+        return false;
```
[the remark this answers is not in the document]
Comment (N3SSQwiK) [reply game.js]: Good catch! Fixed in cd6cd44.

`spawn()` now clears `position` and `spawnTick` when returning `false`:

```javascript
// No valid position available (grid is full)
// Clear food state to prevent stale food from being collected
this.position = null;
this.spawnTick = null;
return false;
```

Added test: "spawn clears food state when grid is full" (112 tests now pass).
